src/app/components/qr-scan/qr-scan.component.ts

```diff
@@ -71,7 +71,16 @@ export class QrScanComponent implements OnInit {
       // This is a valid Nano scheme URI
       const url = new URL(resultString);
 
-      if (url.protocol === 'nano:' && this.util.account.isValidAccount(url.pathname)) {
+      // check if QR contains a full URL path
+      if (url.protocol === 'https:') {
+        if (url.pathname === '/import-wallet' && url.hash.slice(1).length) {
+          // wallet import
+          this.router.navigate(['import-wallet'], { fragment: url.hash.slice(1)});
+        } else if (url.pathname === '/import-address-book' && url.hash.slice(1).length) {
+          // address book import
+          this.router.navigate(['import-address-book'], { fragment: url.hash.slice(1)});
+        }
```

Comment:
Here is missing the check for the current domain.
The domain in the QR code has to match with the domain you're on.
Otherwise you could scan `https://thisisnotnault.cc/import-wallet#abc` and it would work which is probably not intended.

Reply:
Okay it's fine as we only take a part of the domain, tampering with Nault isn't possible with this approach.